typed-racket-lib/typed-racket/base-env/prims-contract.rkt

```diff
@@ -250,9 +251,17 @@
 ;;  make-predicate
 ;;  cast
 
-;; Helper to construct syntax for contract definitions
+;; Helpers to construct syntax for contract definitions
+;; make-contract-def-rhs : Type-Stx Boolean Boolean -> Syntax
 (define (make-contract-def-rhs type flat? maker?)
-  (contract-def-property #'#f `#s(contract-def ,type ,flat? ,maker? untyped)))
+  (define contract-def `#s(contract-def ,type ,flat? ,maker? untyped))
+  (contract-def-property #'#f (λ () contract-def)))
+
+;; make-contract-def-rhs/from-typed : Id Boolean Boolean -> Syntax
+(define (make-contract-def-rhs/from-typed id flat? maker?)
+  (contract-def-property
+   #'#f
+   (λ () `#s(contract-def ,(or (cast-table-ref id) #'Any) ,flat? ,maker? typed))))
```

Comment:
```
Internal Typechecker Error: contract-def-property: thunk called too early
  This should only be called after the type-checking pass has finished.
while typechecking:
here
originally:
here
  context...:
   /Users/Alex/Documents/GitHub/typed-racket/typed-racket-lib/typed-racket/utils/tc-utils.rkt:270:0: int-err
   /Users/Alex/Documents/GitHub/typed-racket/typed-racket-lib/typed-racket/base-env/prims-contract.rkt:268:3
   /Users/Alex/Documents/GitHub/typed-racket/typed-racket-lib/typed-racket/private/type-contract.rkt:82:0: generate-contract-def
   /Users/Alex/Documents/GitHub/typed-racket/typed-racket-lib/typed-racket/private/type-contract.rkt:195:0: change-contract-fixups
   /Users/Alex/Documents/GitHub/typed-racket/typed-racket-lib/typed-racket/typecheck/toplevel-trampoline.rkt:108:24: for-loop
   dots-loop
   /applications/racket/2016-05-23/tr/Racket v6.5.0.5/collects/racket/private/more-scheme.rkt:148:2: call-with-break-parameterization
   .../more-scheme.rkt:261:28
   /applications/racket/2016-05-23/tr/Racket v6.5.0.5/share/pkgs/sandbox-lib/racket/sandbox.rkt:843:5: loop
```
It seems like it's going though `generate-contract-def` and `change-contract-fixups`, which are supposed to be for the contract-generation phase, I think. Is there any chance that the contract-generation phase is happening before type-checking is finished?